### PR TITLE
feat: add "gap" option for cubemap/cubestrip

### PR DESCRIPTION
**Pull request by WoodNeck**
## Issue
#357

## Details
|gap: 0|gap: 3|
|---|---|
|다운로드 (1)|다운로드|
|다운로드 (3)|다운로드 (2)|

Any suggestions for the option name is appreciated.

### CODE REVIEW

File: demo/examples/panoviewer/projection-type/cubemap_image.html [hunk 1]
```diff
@@ -64,7 +64,10 @@ <h2>Support variety cubemap format</h2>
 
     var panoViewer = new PanoViewer(container, {
         projectionType: "cubemap",
-        image: "../../img/test_cube_3x2_RLUDFB.jpg"
+        image: "../../img/test_cube_3x2_RLUDFB.jpg",
+        cubemapConfig: {
+          gap: 3
```
Comment (mixed) [inline demo/examples/panoviewer/projection-type/cubemap_image.html:69]: Is the gap a value that the user should properly set?

File: src/PanoViewer/PanoViewer.ts [hunk 2]
```diff
@@ -335,7 +335,8 @@ class PanoViewer extends Component<PanoViewerEvent> {
         tileConfig: {
           flipHorizontal: false,
           rotation: 0
-        }
+        },
+        gap: 0
```
Comment (mixed) [inline src/PanoViewer/PanoViewer.ts:339]: Don't you need to add a description to the API?

**Review by mixed — APPROVED** [review]
LGTM

File: demo/examples/panoviewer/projection-type/cubemap_image.html (hunk 1 above)
Comment (WoodNeck) [reply demo/examples/panoviewer/projection-type/cubemap_image.html:69]: I think we can't find value automatically as it's texture issue.

File: src/PanoViewer/PanoViewer.ts (hunk 2 above)
Comment (WoodNeck) [reply src/PanoViewer/PanoViewer.ts:339]: Yeah I forgot that, I'll add one.

**Review by daybrush — APPROVED** [review]
I think it's better to set it to a negative number, as positive numbers mean more margins.

**Comment by WoodNeck** [comment]
@daybrush 
What do you mean? It is true that we're increasing margins.

**Comment by daybrush** [comment]
@WoodNeck 
Does increasing the margin mean that there are more pillars in between?

Doesn't the pillar disappear by removing the margin?

I understand that the shrink code also removes the margin.

**Comment by WoodNeck** [comment]
@daybrush 
image
This is the margin I mean

**Comment by daybrush** [comment]
@WoodNeck 

I dom't know what is in the image above. Isn't it this?


* margin: 0



* margin: 2 (The margin I think is -2).
 




Is the margin option that you want to add is the margin that the texture has? Is the margin to be set?

**Comment by WoodNeck** [comment]
@daybrush 
The margin is applied to the texture not the final image to trim the side of each tiles.

image

I'm not referring a CSS box model but a typical term "margin", so it doesn't have to be a negative number because it's just inside the image.

Should I rather have to use word like `trim` or something?

**Comment by daybrush** [comment]
@WoodNeck 


The word `trim` seems to be good.

**Comment by WoodNeck** [comment]
Renamed `margin` to `trim`

**Pull request merged**